Frontend/ui-customer/src/Components/IndividualRideCard/View.purs

```diff
@@ -365,14 +365,15 @@ viewDetailsAndRepeatRide push state =
       , height WRAP_CONTENT
       , text $ getString VIEW_DETAILS
       , color Color.blue900
-      , padding $ Padding 26 18 50 3
+      , if getValueToLocalStore IS_SERVICEABLE == "true" then padding $ Padding 26 18 50 3 else padding $ Padding 26 18 26 3
```

Comment:
Why is SharedPref variable required? Can we use state variables itself for the same?

Reply:
as the individual ride component are show using a  PrestoList, I tried settings value of state variable of a component, based on value of state variable of the screen the component is in, by specifying it in a transformer in Frontend/ui-customer/src/Screens/CustomerUtils/MyRidesScreen/Controller.purs:208
```haskell
myRideListTransformer :: MyRidesScreenState -> Array RideBookingRes -> Array IndividualRideCardState
myRideListTransformer state listRes = filter (\item -> (item.status == "COMPLETED" || item.status == "CANCELLED")) (map (\(RideBookingRes ride) ->
  let
    fares = getFares ride.fareBreakup
    ...
  , isSrcServiceable: state.data.isSrcServiceable
}) (listRes))
```
then access it in component like  (./src/Components/IndividualRideCard/View.purs:393)
```haskell
viewDetailsAndRepeatRideShimmer :: forall w. IndividualRideCardState -> PrestoDOM (Effect Unit) w
viewDetailsAndRepeatRideShimmer state =
  linearLayout
  [ width MATCH_PARENT
  , height WRAP_CONTENT
  , gravity CENTER_HORIZONTAL
  , margin $ MarginTop 15
  ][  sfl $ textView (
      [ width WRAP_CONTENT
      , height WRAP_CONTENT
      , text $ getString VIEW_DETAILS
      , color Color.grey900
      , background Color.grey900
      , padding $ Padding 10 3 10 3
      , PrestoList.visibilityHolder "shimmerVisibility"
      ] <> FontStyle.body1 LanguageStyle)
    , linearLayout 
      [ width $ V 1
      , height if os == "IOS" then (V 10) else MATCH_PARENT
      , margin $ MarginHorizontal 40 40
      , background Color.grey900
      , visibility if state.isSrcServiceable then VISIBLE else GONE
      ][]
    , sfl $ textView (
      [ width WRAP_CONTENT
      , height WRAP_CONTENT
      , text $ getString REPEAT_RIDE
      , PrestoList.visibilityHolder "shimmerVisibility"
      , color Color.grey900
      , background Color.grey900
      , padding $ Padding 10 3 10 3
      , visibility if state.isSrcServiceable then VISIBLE else GONE
      ] <> FontStyle.body1 LanguageStyle)
  ]
```
but the value is not  updating,

Reply:
update: I have fixed this by updating value of dummy card value in my ride screen handler